src/app/reporting/Engine.h

```diff
@@ -66,10 +67,12 @@ class Engine : public DataModel::ProviderChangeListener
     /**
      * Initializes the reporting engine. Should only be called once.
      *
+     * @param[in] A pointer to EventManagement. Use the global one by default.
```

Comment:
Do we need to preserve that "default" behavior at all?

Reply:
If we are okay with changing the interface, we can remove the default. I can only see InteractionModelEngine calling this init and it seems the pointer is properly added here. 

Remove the default and verify that the input argument is not a nullptr.